regression-tests/pure2-bugfix-for-discard-precedence.cpp2

```diff
@@ -1,16 +1,11 @@
 quantity: type = {
   number: i32;
-  operator=: (out this, i: std::in_place_t, x: i32) = {
-    number = x;
-    _      = i;
-  }
-  operator+=: (inout this, that) -> forward quantity = {
-    number += that.number;
-    return this;
-  }
+  operator=: (out this, x: i32) = number    = x;
+  operator+: (inout this, that) -> quantity = quantity(number + that.number);
 }
 
-main: () = {
-  x: quantity = (std::in_place, 1729);
-  x += x;
+main: (args) = {
+  x: quantity = (1729);
+  _           = x + x; // Not `(void) x + x`; would attempt to add a `void` to `x`.
+  _           = args;  // Not `void(args)`; would attempt to declare `args` with `void` type.
```

Comment:
This scary syntax can be used to smuggle declarations:
<https://cpp2.godbolt.org/z/EsGKsThz3>
```Cpp2
main: () = i32(x);
```
```C++
auto main() -> int { cpp2::i32(x);  }
```
```output
main.cpp2:1:32: warning: unused variable 'x' [-Wunused-variable]
    1 | auto main() -> int { cpp2::i32(x);  }
      |                                ^
1 warning generated.
Program returned: 0
```

Reply:
Delightful! FWIW `Wparentheses` would flag that.